plugins/connection/lxd.py

```diff
@@ -125,6 +161,25 @@ def exec_command(self, cmd, in_data=None, sudoable=True):
 
         return process.returncode, stdout, stderr
 
+    def _get_remote_uid_gid(self) -> tuple[int, int]:
+        """Get the user and group id of 'remote_user' from the instance."""
+
+        rc, uid_out, err = self.exec_command("/bin/id -u")
+        if rc != 0:
+            raise AnsibleError(
+                f"Failed to get remote uid for user {self.get_option('remote_user')}: {err}"
+            )
+        uid = uid_out.strip()
+
+        rc, gid_out, err = self.exec_command("/bin/id -g")
+        if rc != 0:
+            raise AnsibleError(
+                f"Failed to get remote gid for user {self.get_option('remote_user')}: {err}"
+            )
+        gid = gid_out.strip()
+
+        return int(uid), int(gid)
```

Comment:
These commands will run with the same uid/gid as the python process running the plugin, right? So why running external commands instead of using `os.getuid()` and `os.getgid()`?

Reply:
Got it. In that case I think you need to stick to the `id` command indeed. Thanks for clarifying.